### PR TITLE
Issue #1163 - Convert BashPrimaryExpressions to use template

### PR DESCRIPTION
**Pull request by Mailkov**
This is WIP 

-----
IA Page: https://duck.co/ia/view/bash_primary_expressions

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata post coming soon

File: lib/DDG/Goodie/BashPrimaryExpressions.pm [hunk 1]
```diff
@@ -78,24 +78,49 @@ handle remainder => sub {
...
 	}
 
 	$text_output =~ s/ARG2/$right_arg/g;
-	$html_output =~ s/ARG2/<pre>$html_right_arg<\/pre>/g;
-
-	my $intro = "The Bash expression <pre>" . html_enc($_) . "</pre> results to";
-	return "$intro $text_output.", html => "$intro $html_output.", heading => html_enc($_) . " (Bash)";
+	
+	my @results;   
+    foreach my $el(@{$if_description{$op}}) {
```
Comment (moollaza) [inline lib/DDG/Goodie/BashPrimaryExpressions.pm]: Might be better to use `map` here -- then you don't need to `push`

File: lib/DDG/Goodie/BashPrimaryExpressions.pm [hunk 2]
```diff
@@ -22,43 +22,43 @@ zci answer_type => 'expression_description';
...
+	'-lt' => ["true if ", "ARG1", " is numerically less than ", "ARG2"],
+	'-le' => ["true if ", "ARG1", " is numerically less than or equal to ", "ARG2"],
+	'-gt' => ["true if ", "ARG1", " is numerically greater than ", "ARG2"],
+	'-ge' => ["true if ", "ARG1", " is numerically greater than or equal to ", "ARG2"],
+	'-nt' => ["true if ", "ARG1", " has been changed more recently than ", "ARG2", " or if ", "ARG1", " exists and ", "ARG2", " does not"],
+	'-ot' => ["true if ", "ARG1", " is older than ", "ARG2", " or ", "ARG2", " exists and ", "ARG1", " does not"],
+	'-ef' => ["true if ", "ARG1", " and ", "ARG2", " refer to the same device and inode numbers"]
 );
 
 handle remainder => sub {
```
Comment (mintsoft) [inline lib/DDG/Goodie/BashPrimaryExpressions.pm]: Looks like I screwed up the indentation just below here, mind fixing it while you're here?

File: lib/DDG/Goodie/BashPrimaryExpressions.pm [hunk 3]
```diff
@@ -78,24 +78,49 @@ handle remainder => sub {
...
-	    $html_output =~ s/ARG1/<pre>$html_left_arg<\/pre>/g;
 	}
 
 	$text_output =~ s/ARG2/$right_arg/g;
-	$html_output =~ s/ARG2/<pre>$html_right_arg<\/pre>/g;
-
-	my $intro = "The Bash expression <pre>" . html_enc($_) . "</pre> results to";
-	return "$intro $text_output.", html => "$intro $html_output.", heading => html_enc($_) . " (Bash)";
+	
+	my @results;
```
Comment (mintsoft) [inline lib/DDG/Goodie/BashPrimaryExpressions.pm]: Hard-tabs here too!

File: lib/DDG/Goodie/BashPrimaryExpressions.pm (hunk 1 above)
Comment (mintsoft) [inline lib/DDG/Goodie/BashPrimaryExpressions.pm]: There should only ever be one result though, it seems a little odd to using an array here to me

Comment (Mailkov) [inline lib/DDG/Goodie/BashPrimaryExpressions.pm]: @mintsoft I changed `%if_description` now `$if_description{$op}` contains an array of strings

File: lib/DDG/Goodie/BashPrimaryExpressions.pm (hunk 2 above)
Comment (Mailkov) [inline lib/DDG/Goodie/BashPrimaryExpressions.pm]: @mintsoft Sure!

File: lib/DDG/Goodie/BashPrimaryExpressions.pm (hunk 3 above)
Comment (Mailkov) [inline lib/DDG/Goodie/BashPrimaryExpressions.pm]: @mintsoft Yes, this is a WIP so I have not checked everything

File: share/goodie/bash_primary_expressions/content.handlebars [hunk 4]
```diff
@@ -0,0 +1,8 @@
+The Bash expression <pre>{{intro}}</pre>
```
Comment (moollaza) [inline share/goodie/bash_primary_expressions/content.handlebars]: Looks like we're missing the "results to" text?

**Comment by moollaza** [comment]
@Mailkov is this done? It looks like the conversion to templates is complete. I'd like to see it use a title/subtitle approach but we can merge and make an issue for that. 

It's not using HTML anymore which is the goal :)

**Pull request merged**